### DISCUSSION TITLE
Build older python package - 3.9

### DISCUSSION THREAD

**Discussion by Matshec** [termux/termux-packages#9498] [Q&A] [answered]
Hi, 
I want to build python 3.9 for termux since only available version right now is 3.10.  The reason is  I would like to run homeassistant on termux but python 3.10 is not supported by it yet. 

My problem is:
Trying to build it on my laptop I get the message
`ERROR: Wrong NDK version - we need 21d
`
I am using official docker image,
I am checked out on git to older commit: 04771833e7384de4e33d3279714a29c242db68a0 when python3.9 was the main one 

Can you give me some tips how to solve this please?

**Comment by Exanoff12** [comment]
try this and see if it works

1. first uninstall python latest (`pkg uninstall python -y`)
2. install python 3.9 (`apt install python3.9 -y`)
3. try building now

hopefully this helped you !
make sure to update and upgrade (`yes | pkg update && yes | pkg upgrade`) before executing

**Reply by KianBrose** [reply]
It doesn't

**Reply by sylirre** [reply]
`termux-packages` provides only `python` (the latest) and `python2` (v2.7) packages. Instructions won't work unless one has a TUR repo configured.

**Comment by sylirre** [comment, marked as answer]
We don't build specific package versions on request. This task is for https://github.com/termux-user-repository/tur (TUR - Termux User Repo) and its maintainers. Python 3.9 is already available in this repository.

How to:
```
pkg install tur-repo
pkg update
pkg install python3.9
```

**Reply by xdevfaheem** [reply]
its telling python not installed when python3.9 is installed

**Reply by sk0kanik** [reply]
> its telling python not installed when python3.9 is installed

```
~ $ python3.9 -V
Python 3.9.16
```

**Reply by xdevfaheem** [reply]
Yes, but when i start python
~python

It Show Python Not Installed install it by
pkg install python

**Reply by sk0kanik** [reply]
Yes, default python (currently 3.11) is not installed. You can use ``python3.9`` instead of ``python`` or create a symlink named "python" pointed to "python3.9"
``ln -s python3.9 $PREFIX/bin/python``.

**Reply by xdevfaheem** [reply]
Thank You, I Successfully pointed python3.9 to python. You Saved Me. But How can i install pip here

**Reply by xdevfaheem** [reply]
Please Reply ASAP

**Reply by T-Dynamos** [reply]
Pip is installed already. Run with `python3.9 -m pip`

**Reply by xdevfaheem** [reply]
Thank You So Much Guys You Saved me.

I'm Doing All Of These Because numpy is not supporting < 3.11.
And For Scipy too, These are All Important for My Machine Learning and Deep Learning Projects.

Once Again Thank You Very Much.

**Reply by xdevfaheem** [reply]
One More to Ask, How can I Use python3.9 pip as default.

Because it uses pip from python3.11 when i start pip3 install numpy.

**Reply by T-Dynamos** [reply]
Yes you may make a symbolic link
`ln -s $PREFIX/bin/pip3.9 $PREFIX/bin/pip`
(You may make backup of `$PREFIX/bin/pip`)
